packages/contracts-bedrock/src/L1/SuperchainConfig.sol

```diff
@@ -0,0 +1,280 @@
+            bytes32 sequencerHash = Hashing.hashSequencerKeys(_sequencers[i]);
+            allowedSequencers[sequencerHash] = true;
+            emit ConfigUpdate(UpdateType.ADD_SEQUENCER, abi.encode(_sequencers[i]));
+        }
+    }
+
+    /// @notice Returns an address stored in an arbitrary storage slot.
+    ///         These storage slots decouple the storage layout from
+    ///         solc's automation.
+    /// @param _slot The storage slot to retrieve the address from.
+    function _getAddress(bytes32 _slot) internal view returns (address addr_) {
+        assembly {
+            addr_ := sload(_slot)
+        }
+    }
+
+    /// @notice Stores an address in an arbitrary storage slot, `_slot`.
+    /// @param _slot The storage slot to store the address in.
+    /// @param _address The protocol version to store
+    /// @dev WARNING! This function must be used cautiously, as it allows for overwriting addresses
+    ///      in arbitrary storage slots.
+    function _setAddress(bytes32 _slot, address _address) internal {
+        assembly {
+            sstore(_slot, _address)
+        }
+    }
+
+    /// @notice Returns a uint256 stored in an arbitrary storage slot.
+    ///         These storage slots decouple the storage layout from
+    ///         solc's automation.
+    /// @param _slot The storage slot to retrieve the address from.
+    function _getValue(bytes32 _slot) internal view returns (uint256 value_) {
+        assembly {
+            value_ := sload(_slot)
+        }
+    }
+
+    /// @notice Stores a value in an arbitrary storage slot, `_slot`.
+    /// @param _slot The storage slot to store the address in.
+    /// @param _value The protocol version to store
+    /// @dev WARNING! This function must be used cautiously, as it allows for overwriting values
+    ///      in arbitrary storage slots.
+    function _setValue(bytes32 _slot, uint256 _value) internal {
+        assembly {
+            sstore(_slot, _value)
+        }
+    }
+
+    /// @notice Getter for the systemOwner address.
+    function systemOwner() public view returns (address systemOwner_) {
+        systemOwner_ = _getAddress(SYSTEM_OWNER_SLOT);
+    }
+
+    /// @notice Getter for the initiator address.
+    function initiator() public view returns (address initiator_) {
+        initiator_ = _getAddress(INITIATOR_SLOT);
+    }
+
+    /// @notice Getter for the vetoer address.
+    function vetoer() public view returns (address vetoer_) {
+        vetoer_ = _getAddress(VETOER_SLOT);
+    }
+
+    /// @notice Getter for the guardian address.
+    function guardian() public view returns (address guardian_) {
+        guardian_ = _getAddress(GUARDIAN_SLOT);
+    }
+
+    /// @notice Getter for the delay address.
+    function delay() public view returns (uint256 delay_) {
+        // We do some casting rather than define a new getter.
+        delay_ = uint256(uint160(_getAddress(DELAY_SLOT)));
```

Comment:
This seems bug prone if the delay is more than 160 bits, unlikely in practice but imo these sorts of shortcuts create the habits that lead to bugs

Reply:
hmmm, looks like I already have a getValue() I can use there anyways.